### PR TITLE
Burrower fixes

### LINKED ISSUE #5000: Burrower is invisible in the hivemind vision
## Description
You can not see Burrowers through wall with hivemind vision (they are visible if they are burrowed), you can still see HP and plasma however

### PR DESCRIPTION
**Pull request by blueDev2**
## About the PR

Bug fixes
Resolves #5000
Fixes https://github.com/RMC-14/RMC-14/issues/5012

## Requirements

- [ ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [ ] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- fix: Fixed tunnels renaming, they are now only renamable by a xeno of drone evolution and from the same hive
- fix: Tunnels' default name is now in parity
- fix: Mapped tunnels are properly named and associated to the hive
- fix: Fixed xenos going into tunnels while unable to move
- fix: Fixed involuntary regurgitation from leaving marines in tunnels
- fix: Fixed burrowers being moved from explosives
- fix: Fixed burrowers from being set on fire while burrowed
- fix: Fixed burrower stomp range from being too large, (range of 4 now, used to be a range of 6)
- fix: Add Tunnels to the tactical maps
- fix: Burrowers now are visible to fellow xenos in night vision
- fix: Xenos cannot drop items while in tunnels
- fix: Xenos cannot use actions while in tunnels

### CODE REVIEW

File: Content.Shared/_RMC14/Xenonids/Construction/Tunnel/TunnelRenamerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Content.Shared._RMC14.Xenonids.Construction.Tunnel;
+[RegisterComponent]
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Xenonids/Construction/Tunnel/TunnelRenamerComponent.cs:8]: This needs to be a networked component as well or the client will never know about it

File: Content.Server/_RMC14/Xenonids/Burrow/XenoBurrowSystem.cs [hunk 2]
```diff
@@ -290,6 +290,8 @@ private void SetBurrow(Entity<XenoBurrowComponent> xenoBurrower, bool active)
                 Dirty(action.Id, actComp);
             }
```
Comment (DrSmugleaf) [inline Content.Server/_RMC14/Xenonids/Burrow/XenoBurrowSystem.cs:292]: This needs to call physics system try set body type kinematic controller after unanchor, or the xeno will be launched to the moon by explosions as happened with the queen and defenders

File: Resources/Prototypes/_RMC14/Entities/Mobs/Xeno/lesser_drone.yml [hunk 3]
```diff
@@ -72,6 +72,7 @@
       - XenoParasite
       - XenoEgg
       - XenoFruit
+  - type: TunnelRenamer
```
Comment (DrSmugleaf) [inline Resources/Prototypes/_RMC14/Entities/Mobs/Xeno/lesser_drone.yml:75]: Is the lesser also supposed to be able to do this?

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: Resources/Prototypes/_RMC14/Entities/Mobs/Xeno/lesser_drone.yml (hunk 3 above)
Comment (blueDev2) [reply Resources/Prototypes/_RMC14/Entities/Mobs/Xeno/lesser_drone.yml:75]: Yup
https://github.com/cmss13-devs/cmss13/blob/570fc8f2777381f4a88b4029da2540c72f91b253/code/modules/mob/living/carbon/xenomorph/castes/lesser_drone.dm#L74

File: Content.Server/_RMC14/Xenonids/Burrow/XenoBurrowSystem.cs [hunk 4]
```diff
@@ -290,6 +292,9 @@ private void SetBurrow(Entity<XenoBurrowComponent> xenoBurrower, bool active)
...
 
+            _transform.Unanchor(ent);
+            _physics.SetBodyType(ent, BodyType.Kinematic);
```
Comment (DrSmugleaf) [inline Content.Server/_RMC14/Xenonids/Burrow/XenoBurrowSystem.cs:296]:
```suggestion
            _physics.SetBodyType(ent, BodyType.KinematicController);
```

**Review by DrSmugleaf — DISMISSED** [review]

**Pull request merged**